tap_eloqua/sync.py

```diff
@@ -78,24 +88,33 @@ def stream_export(client, state, catalog, stream_name, sync_id, updated_at_field
             stream_name,
             offset,
             bulk_page_size))
-        data = client.get(
-            '/api/bulk/2.0/syncs/{}/data'.format(sync_id),
-            params={
-                'limit': bulk_page_size,
-                'offset': offset
-            },
-            endpoint='export_data')
-        has_more = data['hasMore']
-        offset += bulk_page_size
-
-        if 'items' in data and data['items']:
-            records = map(transform_export_row, data['items'])
-            persist_records(catalog, stream_name, records)
+        with metrics.job_timer('export_page'):
+            stream = client.get(
+                '/api/bulk/2.0/syncs/{}/data'.format(sync_id),
+                params={
+                    'limit': bulk_page_size,
+                    'offset': offset
+                },
+                stream_csv=True,
+                endpoint='export_data')
+            offset += bulk_page_size
+
+            records_stream = (line.decode('utf-8') for line in stream.iter_lines())
+            records = csv.DictReader(records_stream)
+            records_transformed = map(transform_export_row, records)
+
+            count, max_page_updated_at = persist_records(
+                catalog,
+                stream_name,
+                records_transformed,
+                updated_at_field=updated_at_field)
 
-            max_page_updated_at = max(map(lambda x: x[updated_at_field], data['items']))
             if max_updated_at is None or max_page_updated_at > max_updated_at:
                 max_updated_at = max_page_updated_at
 
+            if count < bulk_page_size:
+                break
```

Comment:
It took me a minute to understand why this `break` is here. Can you either change this to set `has_more` to False rather than breaking? Right now, the while is effectively a `while True`.